### PR TITLE
device: Add deactivate method

### PR DESCRIPTION
**Pull request by mbalamat**
Add device deactivate method

### CODE REVIEW

File: tests/integration/models/device.spec.js
Comment (sradevski) [inline tests/integration/models/device.spec.js:1098]: Can you add a test case where it does succeed, just for completeness?

**Review by sradevski — COMMENTED** [review]
You can also PR this in the dashboard after it's merged 😄

File: tests/integration/models/device.spec.js
Comment (mbalamat) [reply tests/integration/models/device.spec.js:1098]: @sradevski I thought the same thing though this involves billing so the user that runs the device spec tests should have billing info attached to them, which it doesn't. I asked [here](https://www.flowdock.com/app/rulemotion/r-beginners/threads/Dco-wiN7GyqHKk1_ZTvXBIpq_Ti).

Comment (sradevski) [reply tests/integration/models/device.spec.js:1098]: I see, I'll let Thodoris approve it in case he has something to add, otherwise LGTM. If it's urgent, feel free to self-certify and we can tackle this later.

Comment (thgreasi) [reply tests/integration/models/device.spec.js:1098]: @mbalamat let's move this last under the `given an application > given a single offline device > [contained scenario]`.
Then let's also add a deactivation test case for the device in that context that checks the status code and error message that the API gives back (probably something about not being able to deactivate devices in starter apps.
Hint to check the status code: `expect(error).to.have.property('statusCode', 401);`

Then lastly lets also add a test case for a failing deactivation under the `given an application > given an online device` that should give "401" & "Devices must be offline in order to be deactivated".

Even though these are just failing test cases, the current two do not test anything after the `device.get()`, so we are never checking whether the API knows the `is_active` field or not.

**Review by thgreasi — COMMENTED** [review]
Other than the extra failing test cases LGTM :+1: 
Great first SDK PR. Nice digging.

**Review by thgreasi — APPROVED** [review]
Would prefer async await in the test but this is already great :+1:

**Pull request merged**
